### ISSUE TITLE
Favourites have really long name due to using PDF name 

### ISSUE THREAD

**Issue by AntonioCS** [sumatrapdfreader/sumatrapdf#2236]
Hey,
I have multiple documents opened and the book I'm reading I added a favourite.
The issue is that the favourite had the full name of the book I was viewing and then the name I gave it.
So it was like
```
<full name of the pdf I was reading>: <name I gave bookmark>
```
I can understand that this is to differ when I select another pdf but with really long pdf names this makes it really hard to keep track of things.
Example:
![image](https://user-images.githubusercontent.com/1086401/140274851-115a7b74-ab19-44e7-87d9-b83b37ff66d2.png)
I don't think this is very useful. This pane takes like 30%+ of screen for me to see the actual name that I gave this "favourite".

**Comment by AntonioCS** [comment]
Also to note, that I am unable to change the name once it's given. I'm not sure if in previous versions I was able to edit this or not, but right now I only have an option to remove it.

**Comment by GitHubRulesOK** [comment]
There was a recent change in pre-release to improve long lengths by culling some of the middle text but there will be those that want the full name wrapped, as ever there has to be compromise.

As to editing you can alter the entry in Advanced settings using the Notepad viewer

I keep the window narrow and hover to see the long name in tooltip
![image](https://user-images.githubusercontent.com/12754913/140312681-4b30c8b9-b489-44bc-93f1-810f01e8714f.png)

**Comment by AntonioCS** [comment]
@GitHubRulesOK so... sumatra keeps states on ALL files??
I'm looking at the settings file and the `FileStates` is a HUGE list of files...
This can't be efficient right? 
I'm seeing files there that are long gone from sumatra..

**Comment by AntonioCS** [comment]
Also, can't there be a setting somewhere? Where people just have an option to configure the favourite syntax?
Like a input somewhere that has predefined keywords that mean something: 
```
%filename% %favname%
```
Or something similar. If there are a lot of people discussing this a simple option like t his might do the trick to please the majority. You just let people chose which one they like.

**Comment by terax6669** [comment]
We don't have to change much to improve the situation.

![image](https://user-images.githubusercontent.com/19204073/168442174-2af49d93-0ed4-4f8a-a892-30bd4390371c.png)

1. Remove `.pdf` from the PDF title.
2. Add possibility to rename the PDF title. Optional, but very helpful. I keep my filenames very descriptive and they take up too much space in the example.
3. As a last resort: detect overflow and switch from single combined line (filename + bookmark name) to dropdown.

**Comment by kjk** [comment]
I agree favorites should be improved. In general: fast navigation in the document or between documents.

I'm thinking that favorites should work more like Command Palette: instead of "always visible" tree of favorites, it should be "invoke favorites window with keyboard shortcut/command, choose bookmark, close window"

The window would be bigger, solving the issue of long names.

There would be search filtering functionality (like in Command Palette) to quickly find entry to go to.

Also related are even faster way of quickly jumping between points in the document. https://github.com/ahrm/sioyek has some ideas I could reuse.

**Comment by Geobert** [comment]
I think this is related to https://github.com/sumatrapdfreader/sumatrapdf/issues/829

**Comment by kjk** [comment]
Fixed together with #829. The favorites pane (and menu) previously showed a single-favorite entry as `<filename> : <your favorite's name>`, so a long file name pushed the name you gave the favorite out of view. It now shows the favorite's name first, then the file name, so your description is always visible. This will be in the next pre-release / 3.7.
